torchaudio/models/decoder/_ctc_decoder.py

```diff
@@ -9,13 +11,13 @@
 import torchaudio
 from torchaudio.utils import download_asset
 
-try:
-    # We prioritize the version from upstream flashlight here.
-    # This will allow applications that use the upstream flashlight
-    # alongside torchaudio.
+
+# We prioritize the version from upstream flashlight here.
+# This will allow applications that use the upstream flashlight
+# alongside torchaudio.
+if torchaudio._internal.module_utils.is_module_available("flashlight"):
```

Comment:
Flashlight is a namespace package FYI -- `flashlight.lib.sequence` is another package but doesn't have any of what you need. You might want to check if `flashlight.lib.text` is available.

Reply:
The underlying implementation [`importlib.util.find_spec`](https://docs.python.org/3/library/importlib.html#importlib.util.find_spec) only works for top-level module, if the target module is not installed.
To inspect "flashlight.lib.text", it actually attempt to import the "flashlight.lib", so if flashlight is not available it fails with exception like the following.

The goal here is to inspect the module's existence without importing it (so that we won't have symbols from flashlight in case the initialization logic decides not to import from there), so we make assumption that (we ask users to) install when installing flashlight, decoder submodule is also installed, which I think is a reasonable assumption.

```
>>> torchaudio._internal.module_utils.is_module_available("ujktltvbhbvkrdnuvukjhejdhivetkeh")
False
>>> torchaudio._internal.module_utils.is_module_available("ujktltvbhbvkrdnuvukjhejdhivetkeh.foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/moto/Development/torchaudio/torchaudio/_internal/module_utils.py", line 14, in is_module_available
    return all(importlib.util.find_spec(m) is not None for m in modules)
  File "/Users/moto/Development/torchaudio/torchaudio/_internal/module_utils.py", line 14, in <genexpr>
    return all(importlib.util.find_spec(m) is not None for m in modules)
  File "/Users/moto/miniconda3/lib/python3.9/importlib/util.py", line 94, in find_spec
    parent = __import__(parent_name, fromlist=['__path__'])
ModuleNotFoundError: No module named 'ujktltvbhbvkrdnuvukjhejdhivetkeh'
>>>
```

Reply:
Ah, I didn't know `is_module_available` used `find_spec` -- makes sense 👍